packages/engine/src/renderer/WebGLRendererSystem.ts

```diff
@@ -255,13 +265,18 @@ export class EngineRenderer {
   calculateMovingAverage = (delta: number): number => {
     this.averageFrameTime =
       (this.averageFrameTime * this.timeSamples.length + delta - this.timeSamples[this.index]) / this.timeSamples.length
-
     this.timeSamples[this.index] = delta
     this.index = (this.index + 1) % this.timeSamples.length
-
     return this.averageFrameTime
   }
 
+  onUpdateMovingAverage(newValue: number): number {
```

Comment:
I think we should just put the moving average logic in a utility class like the one I provided; it will likely be used in multiple places, and it's very self-contained